test/engine.go

```diff
@@ -433,10 +442,17 @@ func (e *engine) AddCounter(from, to frontend.Tag) {
 	// do nothing, we don't measure constraints with the test engine
 }
 
-func (e *engine) toBigInt(i1 frontend.Variable) big.Int {
-	b := utils.FromInterface(i1)
-	b.Mod(&b, e.modulus())
-	return b
+func (e *engine) toBigInt(i1 frontend.Variable) *big.Int {
+	switch vv := i1.(type) {
+	case *big.Int:
```

Comment:
shouldn't we still do a mod reduce if it's a big int? may slow things down (or we could have a fast path with a comparaison first) --> you can have circuit inputs / constants at this stage, so it may trigger weird edge cases if we allow some values in the test engine to not be mod reduced?

Reply:
I tried to rewrite the API methods such that `toBigInt` is only called for method inputs and I always recreate a new `*big.Int` for temp variables and result. The operations and results are always mod reduced.

I also tried to mod reduce the inputs in `toBigInt` method, but as I will be modifying inputs, I started having a few failing edge cases for the integration tests. Another approach would be to allocate a new `*big.Int`, set its value from the input and mod reduce it, but this goes against the goal of this PR which was to prevent unnecessary allocations.